libbpf-rs/tests/common/mod.rs

```diff
@@ -116,9 +126,18 @@ pub fn get_symbol_offset(binary_path: &Path, symbol_name: &str) -> Result<usize,
         }
         if let Some(name) = elf.strtab.get_at(sym.st_name) {
             if name == symbol_name {
-                let sec = &elf.section_headers[sym.st_shndx];
-                let offset = sym.st_value - sec.sh_addr + sec.sh_offset;
-                return Ok(offset as usize);
+                for phdr in elf.program_headers.iter() {
+                    if phdr.p_type != PT_LOAD {
+                        continue;
+                    }
+                    if phdr.p_flags & PF_X == 0 {
+                        continue;
+                    }
```

Comment:
Does this check add any value? Wouldn't things pan out even if the section were not executable? Probably doesn't matter much in this bit of test code, but it seems superfluous nevertheless.

Reply:
I think it is reasonable to be executable. Otherwise the uprobe attach may also fail I guess. Need more investigation.